### PR TITLE
Add test items for job utils

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
Add test item for job util

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/job/utils_test.go [hunk 1]
```diff
@@ -46,4 +46,21 @@ func TestIsJobFinished(t *testing.T) {
...
+
+	job.Status.Conditions[0].Status = v1.ConditionFalse
+	if IsJobFinished(job) {
+		t.Error("Job was not expected to be finished")
+	}
+
+	job.Status.Conditions[0].Status = v1.ConditionUnknown
+	if IsJobFinished(job) {
+		t.Error("Job was not expected to be finished")
+	}
```
Comment (janetkuo) [inline pkg/controller/job/utils_test.go]: Thanks a lot for the patch. Would you make these tests [table-driven](https://github.com/golang/go/wiki/TableDrivenTests) while you're at it?

**Comment by zhangxiaoyu-zidif** [comment]
Hi， @janetkuo 
Fixed as you conducted. Thanks a lot.
PTAL.

File: pkg/controller/job/utils_test.go [hunk 2]
```diff
@@ -24,26 +24,59 @@ import (
...
-	job := &batch.Job{
-		Status: batch.JobStatus{
-			Conditions: []batch.JobCondition{{
-				Type:   batch.JobComplete,
-				Status: v1.ConditionTrue,
-			}},
+	testCases := map[string]struct {
+		conditionType   batch.JobConditionType
+		conditionStatus v1.ConditionStatus
+		expect          bool
```
Comment (janetkuo) [inline pkg/controller/job/utils_test.go]: I'm not sure what `expect` means? Would you name it `expectJobFinished`, and then return error `if expectJobFinished != IsJobFinished`

Comment (zhangxiaoyu-zidif) [reply pkg/controller/job/utils_test.go]: OK,thanks~

**Comment by zhangxiaoyu-zidif** [comment]
@janetkuo 
PTAL, fixed para.

**Review by janetkuo — APPROVED** [review]

**Pull request merged**